api/sql.ts

```diff
@@ -28,35 +28,183 @@ async function execute(cmd: string, placeholder?: string[]): Promise<any> {
+        profName, 
+        firstName, 
+        lastName, 
+        avgDifficulty, 
+        avgRating, 
+        numRatings, 
+        wouldTakeAgainPercent,
+        id,
+        legacyId) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?)`,
+        [
+          profName.toLowerCase(),
+          firstName,
+          lastName,
+          avgDifficulty.toFixed(1),
+          avgRating.toFixed(1),
+          numRatings.toString(),
+          wouldTakeAgainPercent.toFixed(2),
+          id,
+          legacyId.toString(),
+        ]
+      );
+      console.log(
+        `[SUCCESS] Professor ${profName} has been added to the database.`
+      );
+    } else {
+      console.error(`Professor ${profName} already exists in the database.`);
+    }
+  } catch (err) {
+    console.error(err);
+  }
+}
+
+/**
+ * Updates existing professor data in the SQL database.
+ * @param {Professor} newProfessor See professor interface (/api/Professor.d.ts).
+ */
+export async function updateProf({
+  profName,
+  avgDifficulty,
+  avgRating,
+  numRatings,
+  wouldTakeAgainPercent,
+}: ProfessorUpdate): Promise<void> {
+  try {
+    void execute(
+      `UPDATE rateMyProfessorDB SET
+              avgDifficulty = ?,
+              avgRating = ?,
+              numRatings = ?,
+              wouldTakeAgainPercent = ?,
+              timeAdded = CURRENT_TIMESTAMP
+              WHERE profName = ?`,
+      [
+        avgDifficulty.toFixed(1),
+        avgRating.toFixed(1),
+        numRatings.toString(),
+        wouldTakeAgainPercent.toFixed(2),
+        profName,
+      ]
+    );
+    console.log(`[SUCCESS] Professor ${profName} has been updated.`);
+  } catch (err) {
+    console.error(err);
+  }
+}
+
+/**
+ * Checks if a professor's data in the db is more than 3 months old.
+ * Current expiration date is 3 months, or 7884000 seconds.
+ *
+ * @param {string}  broncoDirectName BroncoDirect name.
+ * @return {Promise<boolean>} True if prof data is 3mo+ old, false otherwise.
  */
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
-async function profSearch(broncoDirectName: string): Promise<void> {
+export async function checkExpiredProfData(
+  broncoDirectName: string
+): Promise<boolean> {
+  const name = broncoDirectName.toLowerCase();
   const result = await execute(
-    'SELECT * FROM `rateMyProfessorDB` WHERE `broncoDirectName` = ?',
+    'SELECT * FROM `rateMyProfessorDB` WHERE `profName` = ? AND TIMESTAMPDIFF(SECOND, `timeAdded`, NOW()) >= 7884000',
+    [name]
+  );
+  if (Object.keys(result).length === 0) {
+    return false;
+  }
+  return true;
+}
+
+/**
+ * Finds a professor in the SQL Database given their BroncoDirect name.
+ * @param {string}  broncoDirectName BroncoDirect name.
+ * @return {Promise<Professor>} JSON of professor data.
+ * Value can be extracted by awaiting function call within an async function.
+ */
+export async function profSearch(broncoDirectName: string): Promise<Professor> {
+  const [result] = await execute(
```

Comment:
What's the reason behind doing `const [result]`? Is it to return the first result?

Reply:
Right now yeah, it's to return the first result

The execute command will return an array of Professor objects, and our backend should only have unique professors in it so we dont have to worry about more than 2 returns (for now). Also, our frontend is only able to handle 1 professor at a time it seems like, so we need to be able to distinguish a null professor and an actual one (so we can't just return the array).

Refactoring the frontend to accomodate that shouldn't be hard, but there's that ig